Get test-graphics project compiling with Mingw-w64

We should probably be building all projects as part of CI, even if we don't run them. That at least keeps them in a state where people can compile them locally and potentially run them. In particular, the **test-graphics** project should be compiled on CI, even though it can't run there (due to lack of a graphical environment, and no user interaction).

[Current build error for **test-graphics** with Mingw-w64](https://app.circleci.com/pipelines/github/lairworks/nas2d-core/1640/workflows/c21cc539-446e-4d0f-aac6-8e7aff481b60/jobs/6557):

> make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror" test-graphics
> x86_64-w64-mingw32-g++ -o .build/debug/testGraphics test-graphics/*.cpp -DGLEW_STATIC -I./ -Werror -Og -g -std=c++17 -Wall -Wextra -Wpedantic -Wzero-as-null-pointer-constant -Wnull-dereference -Wold-style-cast -Wcast-qual -Wcast-align -Wdouble-promotion -Wshadow -Wnon-virtual-dtor -Woverloaded-virtual -Wmissing-declarations -Wmissing-include-dirs -Winvalid-pch -Wmissing-format-attribute -Wredundant-decls -Wno-redundant-decls -I/usr/local/x86_64-w64-mingw32/include/SDL2 -Dmain=SDL_main -Llib  -lnas2d  -lstdc++ -lphysfs -lSDL2_image -lSDL2_mixer -lSDL2_ttf -L/usr/local/x86_64-w64-mingw32/lib -lmingw32 -lSDL2main -lSDL2 -mwindows -Wl,--no-undefined -Wl,--dynamicbase -Wl,--nxcompat -Wl,--high-entropy-va -lm -ldinput8 -ldxguid -ldxerr8 -luser32 -lgdi32 -lwinmm -limm32 -lole32 -loleaut32 -lshell32 -lsetupapi -lversion -luuid -static-libgcc -lglew32 -lopengl32

> <command-line>: error: no previous declaration for 'int SDL_main(int, char**)' [-Werror=missing-declarations]
> test-graphics/main.cpp:22:5: note: in expansion of macro 'main'
>    22 | int main(int /*argc*/, char *argv[])
>       |     ^~~~
> cc1plus: all warnings being treated as errors
> make: *** [makefile:137: .build/debug/testGraphics] Error 1
> make: Target 'test-graphics' not remade because of errors.

Some flags to note:
- `-Dmain=SDL_main`
- `-lSDL2main`

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -18,6 +18,7 @@ commands:
       - run: make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror"
       - run: make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror" test
       - run: make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror" check
+      - run: make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror" test-graphics
       - run: make package
       - store_artifacts:
           path: .build/package/

--- a/makefile
+++ b/makefile
@@ -134,7 +134,7 @@ include $(wildcard $(patsubst $(TESTDIR)/%.cpp,$(TESTINTDIR)/%.d,$(TESTSRCS)))
 test-graphics: $(BUILDDIR)/testGraphics
 $(BUILDDIR)/testGraphics: test-graphics/*.cpp test-graphics/*.h $(OUTPUT)
 	@mkdir -p "${@D}"
-	$(CXX) -o $@ test-graphics/*.cpp $(TESTCPPFLAGS) $(CXXFLAGS) $(TESTLDFLAGS) -lnas2d $(LDLIBS)
+	$(CXX) -o $@ test-graphics/*.cpp $(TESTCPPFLAGS) $(CXXFLAGS) -Umain $(TESTLDFLAGS) -lnas2d $(LDLIBS)
 
 .PHONY: run-test-graphics
 run-test-graphics: | test-graphics

